controllers/texttemplate_controller.go

```diff
@@ -100,6 +102,13 @@ func (r *TextTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 		apimeta.SetStatusCondition(&tt.Status.Conditions, c)
 	}
+	if testing.Testing() {
```

Comment:
This looks a bit odd. I would recommend introducing a `.status.observedGeneration` or another mechanism that allows users to detect that the controller is running and the resource is fully reconciled. We have adopted https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md in most of our controller projects, and it also makes testing easier.

No need to fix this now. More as a follow-up! 😄

Reply:
The problem here is that generation/observedGeneration only gets updated on `spec` changes but not on `metadata` changes. In the case of `TextTemplate`, there is no spec field available that I could use to trigger reconciliation. For `ObjectTemplate` this is a bit different because I can misuse the `interval` field to trigger reconciliation which also causes `generation` updates.

This forces me to use this odd hack for testing... 🙈